pgmpy/readwrite/BIF.py

```diff
@@ -0,0 +1,340 @@
+            probability_block = []
+            for i in probability_block_starts:
+                probability_block_end = self.net.find('}\n',i)
+                probability_block.append(self.net[i:probability_block_end])
+
+            # Creating valid word expression for probability, it is of the format
+            # wor1 | var2 , var3 or var1 var2 var3 or simply var 
+            word_expr = Word(alphanums + '-' + '_') + Suppress(Optional("|")) + Suppress(Optional(","))
+            word_expr2 = Word(initChars=printables, excludeChars=[',',')',' ','(']) + Suppress(Optional(","))
+            # creating an expression for valid numbers, of the format
+            # 1.00 or 1 or 1.00. 0.00 or 9.8e-5 etc
+            num_expr = Word(nums + '-' + '+' + 'e' +'E' +'.') + Suppress(Optional(","))
+            probability_expr = Suppress('probability') + Suppress('(') + OneOrMore(word_expr) + Suppress(')')
+            optional_expr = Suppress('(') + Suppress(OneOrMore(word_expr2)) + Suppress(')')
+            probab_attributes = optional_expr | Suppress('table')
+            cpd_expr = probab_attributes + OneOrMore(num_expr)
+
+            variable_parents = {}
+            variable_cpds = {}
+
+            for block in probability_block:
+                names = probability_expr.searchString(block)
+                names = names[0]
+                variable = names[0]
+                names = names[1:]
+                variable_parents[variable] = names
+                cpds = cpd_expr.searchString(block)
+                arr = [float(j) for i in cpds for j in i]
+                arr = numpy.array(arr)
+                arr = arr.reshape((len(self.variable_states[variable]), 
+                                  arr.size // len(self.variable_states[variable])))
+                variable_cpds[variable] = arr
+
+            return variable_parents, variable_cpds
+
+    def get_variables(self):
+
+        """
+        Returns list of variables of the network
+
+        Example
+        -------------
+        >>> from pgmpy.readwrite import BIFReader
+        >>> reader = BIFReader("bif_test.bif")
+        >>> reader.get_variables()
+        ['light-on','bowel_problem','dog-out','hear-bark','family-out']
+        """
+        try:
+            self.grammar.variable_names
+        except AttributeError:
+            self.grammar = self.Get_grammar(self.network)
+        return self.grammar.variable_names
+
+    def get_states(self):
+
+        """
+        Returns the states of variables present in the network
+
+        Example
+        -----------
+        >>> from pgmpy.readwrite import BIFReader
+        >>> reader = BIFReader("bif_test.bif")
+        >>> reader.get_states()
+        {'bowel-problem': ['true','false'],
+         'dog-out': ['true','false'],
+         'family-out': ['true','false'],
+         'hear-bark': ['true','false'],
+         'light-on': ['true','false']}
+        """
+
+        try:
```

Comment:
@khalibartan Not a very good way to use try and except everywhere. Better do something that makes sure that you have assigned `self.grammar` before using it.